File: backend/app/models/company_settings.py
from datetime import datetime, timezone

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, String, UniqueConstraint
from sqlalchemy.orm import relationship

from app.database import Base


class CompanySettings(Base):
    __tablename__ = "company_settings"
    __table_args__ = (
        UniqueConstraint("company_id", "shipment_prefix", name="uq_company_shipment_prefix"),
        UniqueConstraint("company_id", "invoice_prefix", name="uq_company_invoice_prefix"),
        UniqueConstraint("company_id", "barcode_prefix", name="uq_company_barcode_prefix"),
    )

    id = Column(Integer, primary_key=True, index=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=False, unique=True)
    company_name = Column(String, nullable=False)
    company_logo = Column(String, nullable=True)
    currency = Column(String, nullable=False)
    timezone = Column(String, nullable=False)
    language = Column(String, nullable=False)
    shipment_prefix = Column(String, nullable=False)
    invoice_prefix = Column(String, nullable=False)
    barcode_prefix = Column(String, nullable=False)
    default_cod_percentage = Column(Integer, nullable=False, default=0)
    default_tax_percentage = Column(Integer, nullable=False, default=0)
    sms_provider = Column(String, nullable=True)
    email_provider = Column(String, nullable=True)
    whatsapp_provider = Column(String, nullable=True)
    default_shipment_status = Column(String, nullable=False)
    support_email = Column(String, nullable=True)
    support_phone = Column(String, nullable=True)
    website = Column(String, nullable=True)
    is_active = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)

    company = relationship("Company")

File: frontend/types/companySettings.ts
export type CompanySettings = {
  id: number
  company_id: number
  company_name: string
  currency: string
  timezone: string
  language: string
  shipment_prefix: string
  tracking_prefix: string
  cod_percentage: number
  tax_percentage: number
  shipping_providers: string
  support_phone: string | null
  support_email: string | null
  is_active: boolean
}

export type CompanySettingsPayload = {
  company_id: number
  company_name: string
  company_logo?: string | null
  currency: string
  timezone: string
  language: string
  shipment_prefix: string
  invoice_prefix: string
  barcode_prefix: string
  default_cod_percentage: number
  default_tax_percentage: number
  sms_provider?: string | null
  email_provider?: string | null
  whatsapp_provider?: string | null
  default_shipment_status: string
  support_email?: string | null
  support_phone?: string | null
  website?: string | null
  is_active: boolean
}
